### PR TITLE
sdcc: Update to 4.2

### PR DESCRIPTION
**Pull request by cave-canem**
###### Tested on
macOS 10.9, 10.13
###### Verification 
Have you

- [ ✓ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ✓ ] checked your Portfile with `port lint --nitpick`?
- [ ✓ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by reneeotten** [comment]
@cave-canem please check the CI build failures and rephrase the commit message to follow our guidelines

File: lang/sdcc/Upgrade to 4.2 [hunk 1]
```diff
@@ -4,7 +4,7 @@ PortSystem          1.0
 PortGroup           boost 1.0
 
 name                sdcc
-version             4.0.0
+version             4.2.0
 revision            1
```
Comment (ryandesign) [inline lang/sdcc/Upgrade to 4.2:8]:
```suggestion
revision            0
```

**Comment by reneeotten** [comment]
> port installed sdcc; sudo port lint --nitpick sdcc The following ports are currently installed: sdcc @4.2.0_1 (active) ---> Verifying Portfile for sdcc ---> 0 errors and 0 warnings found.

I restarted the CI builds but they still fail - seems indeed unrelated... so we can probably ignore that. However @ryandesign comments still stands, please make that change.

**Comment by cave-canem** [comment]
Dear …
 
I build this port using clang 5.0 (since it is not so meticulous about C syntax). This port operates correctly on my fairly old computer (MacBook Air 4,1).

P.S.
Please clarify your question because I am not sure I understood it adequately.

**Comment by ryandesign** [comment]
> I restarted the CI builds but they still fail - seems indeed unrelated... so we can probably ignore that.

CI builds fail (and all builds will fail) because the Portfile has been renamed to "Upgrade to 4.2". The filename has to be changed back to "Portfile".

> P.S. Please clarify your question because I am not sure I understood it adequately.

The `revision` should be set to `0` (because you updated the `version`).

I'll make these changes and squash and merge the PR if it builds.

**Pull request merged**